src/utils/macros.rs

```diff
@@ -5,168 +5,184 @@ macro_rules! csi {
     ($( $l:expr ),*) => { concat!("\x1B[", $( $l ),*) };
 }
 
-/// Write a string to standard output whereafter the stdout will be flushed.
+/// Writes a string to the given writer.
 #[doc(hidden)]
 #[macro_export]
-macro_rules! write_cout {
+macro_rules! write_string {
```

Comment:
`write_string` name is kind of misleading. What does `string` in the name mean? Does it mean that we're accepting string? Or does it mean that whatever we give it, string is constructed and then this string will be written to the writer?

When I look at the code, I'd suggest to rename it to the `write_ansi_code`.

Reply:
It can be any type that is displayable, the goal is ansi codes. So let's name it write_ansi_code